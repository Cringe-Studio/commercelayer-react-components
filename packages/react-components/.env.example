# Unit tests
VITE_TEST_CLIENT_ID=
VITE_TEST_ENDPOINT=
VITE_TEST_MARKET_ID=
VITE_TEST_USERNAME=
VITE_TEST_PASSWORD=
VITE_TEST_CLIENT_ID_INTEGRATION=
VITE_TEST_CLIENT_SECRET=
