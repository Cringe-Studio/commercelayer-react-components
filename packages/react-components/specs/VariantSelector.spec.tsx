import React from 'react'
import { VariantSelector } from '../packages/react-components/src'
import renderer from 'react-test-renderer'
import components from '../#config/components'

const propTypes = components.VariantSelector.propTypes

test('<VariantSelector/>', () => {
  expect.assertions(8)
  const component = renderer.create(
    <VariantSelector
      options={[
        {
          label: '6 months',
          code: 'BABYONBU000000E63E746MXX',
        },
      ]}
    />
  )
  const tree = component.toJSON()
  const root = component.toTree()
  const proptypes = root.type['propTypes']
  expect(tree).toMatchSnapshot()
  expect(proptypes.children).toBe(propTypes.children)
  expect(proptypes.placeholder).toBe(propTypes.placeholder)
  expect(proptypes.options).toBe(propTypes.options)
  expect(proptypes.name).toBe(propTypes.name)
  expect(proptypes.type).toBe(propTypes.type)
  expect(proptypes.loader).toBe(propTypes.loader)
  expect(proptypes.skuCode).toBe(propTypes.skuCode)
})

test('<VariantSelector check children />', () => {
  expect.assertions(3)
  const component = renderer.create(
    <VariantSelector
      options={[
        {
          label: '6 months',
          code: 'BABYONBU000000E63E746MXX',
        },
      ]}
    />
  )
  const tree = component.toJSON()
  const root = component.toTree()
  expect(tree).toMatchSnapshot()
  expect(root.props.placeholder).toStrictEqual('Select a variant')
  expect(root.props.type).toStrictEqual('select')
})

test('<VariantSelector with custom children />', () => {
  expect.assertions(5)
  const CustomComponent = (props) => <span>{props.label}</span>
  const component = renderer.create(
    <VariantSelector
      options={[
        {
          label: '6 months',
          code: 'BABYONBU000000E63E746MXX',
        },
      ]}
    >
      {CustomComponent}
    </VariantSelector>
  )
  const tree = component.toJSON()
  const root = component.toTree()
  const rendered = root.rendered
  const parentRendered = root.rendered.rendered
  const childRendered = parentRendered.rendered
  expect(tree).toMatchSnapshot()
  expect(rendered.props.children).toBe(CustomComponent)
  expect(parentRendered.nodeType).toBe('component')
  expect(parentRendered.type).toBe(CustomComponent)
  expect(childRendered.props.handleSelect).not.toBeDefined()
})
