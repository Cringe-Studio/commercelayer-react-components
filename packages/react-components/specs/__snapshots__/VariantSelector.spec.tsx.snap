// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VariantSelector check children /> 1`] = `
<select
  onChange={[Function]}
  value=""
>
  <option
    disabled={false}
  >
    Select a variant
  </option>
</select>
`;

exports[`<VariantSelector with custom children /> 1`] = `<span />`;

exports[`<VariantSelector/> 1`] = `
<select
  onChange={[Function]}
  value=""
>
  <option
    disabled={false}
  >
    Select a variant
  </option>
</select>
`;
